client/my-sites/controller.js

```diff
@@ -349,7 +351,7 @@ module.exports = {
 					if ( waitingNotice ) {
 						notices.removeNotice( waitingNotice );
 					}
-				} else if ( ( currentUser.visible_site_count !== getVisibleSites( getState() ).length ) ) {
```

Comment:
>However, we still have the pesky problem that we can't trust currentUser.visible_site_count to be legit.

Interesting that a difference between `currentUser.visible_site_count` and `getVisibleSites( getState() ).length` is a scenario that is already accounted for. It appears to be a very temporary transitional state, however.

>I think we'll want to follow up on why this mismatch is happening in the endpoints.

Sounds good, I'll clean this up and investigate how `visible_site_count` is gathered by the API